src/safeposix/syscalls/net_calls.rs

```diff
@@ -544,15 +544,13 @@ impl Cage {
 
         if let wrappedfd = fdtable.get(&fd).unwrap() {
             let filedesc_enum = wrappedfd.read().unwrap();
-            match &*filedesc_enum {
-                Socket(socketfd) => {
-                    //do nothing besides check that the fd is a socket
-                    //can't do anything with the fd in here because it is being borrowed in the "if let wrapped fd ="" line
-                    //so we just go to the match statement below -- "match how"
-                },
-                _ => {
-                    return syscall_error(Errno::ENOTSOCK, "netshutdown", "file descriptor is not a socket");
-                }
+
+            //in the code below, do nothing besides check that the fd is a socket
+            //We can't do anything with the fd in here because it is being borrowed in the "if let wrapped fd ="" line
+            //so we just go to the match statement below -- "match how"
+
+            if let Socket(_) = &*filedesc_enum {} else {
+                return syscall_error(Errno::ENOTSOCK, "netshutdown", "file descriptor is not a socket");
             }
```

Comment:
All this logic is just redone in _cleanup_socket (and errnos are even returned) so there's honestly no need to do it here.